addons/mass_mailing/static/src/scss/themes/theme_default.scss

```diff
@@ -103,7 +103,32 @@ div.col:not([align]) {
           width: $width;
           height: $height;
         }
+
+        $size: 3rem;
+
+        &.rounded-circle,
+        &.rounded,
+        &.rounded-0,
+        &.rounded-leaf,
+        &.img-thumbnail,
+        &.shadow {
+            display: inline-block;
+            vertical-align: middle;
+            text-align: center;
+            // fa-1x is not ouput
+            width: $size;
+            height: $size;
+            line-height: $size;
+            @for $i from 2 through 5 {
+                &.fa-#{$i}x {
+                    width: $size + $i;
+                    height: $size + $i;
+                    line-height: $size + $i;
+                }
+            }
+        }
```

Comment:
Shouldn't this be moved from website to somewhere else, like `web_editor` for example, rather than duplicate it in `mass_mailing` ?

@qsm-odoo Is there some particular reason this was done in website specifically ? I suppose we don't want that to apply to the whole backend... I'm not even sure we want that outside of mass_mailing and website... Maybe something like a mixin in `web_editor` that would be called both in `website` and `mass_mailing` ? That would already be better than pure duplication anyway.

Reply:
> Is there some particular reason this was done in website specifically ?

An oversight probably.
Of course not duplicating would be better @visp-odoo: basically, if you hit CTRL-C on multiple lines of code inside your editor, you are doing something wrong :grimacing:

Reply:
@qsm-odoo @dmo-odoo When i'm researching on this task then i found that it is working in 15.0, So i find that in 15.0 this css is written in `mass_mailing` as well as `website`.That is the only reason why i copied it.

Reply:
@visp-odoo So you mean it is in fact duplicated in 15.0 then ?

Reply:
Yes, I have tried creating `mixin` also it is working in `website` but in `mass_mailing` maxin is not found.

Reply:
You mean in 15.0 it doesn't work ? In 16.0 it does, right ? With your current code. If yes then it's not a big deal for 15.0 IMO. The duplication is there, too bad. The more important thing is to not re-introduce the duplication here in 16.0. Like your current solution with a mixin, which is good IMO.

Reply:
No, my current solution is not working in `mass_mailing` when `website` not installed. In mass mailing they show me this error:
```
Error: no mixin named fa-icons-styling
        on line 162:18 of stdin
>>         }

   -----------------^
This error occurred while compiling the bundle 'mass_mailing.assets_mail_themes' containing:
    - /mass_mailing/static/src/scss/themes/theme_basic.scss
    - /mass_mailing/static/src/scss/themes/theme_default.scss
```
I really don't know the reason but its not working in `mass_mailing` .

Reply:
Ah but wait, isn't theme_default.scss only for the plain text template ? You need to find a scss file that is loaded for every template.

Reply:
@dmo-odoo I tried this scss in every file of mass_mailing but i didn't work. It work in only `000.scss` and `001.scss` in mass_mailing. Can we put this css in that file ?

Reply:
If you want to share a mixin between website and mass_mailing, web_editor is probably the right app. You should then put it in a file that is available in all asset bundles, in this case, probably `web_editor.variables.scss` (only the mixin of course, not its use). That is if you need a mixin. I would have imagined there exists a file that share rules between all environments? `web_editor.common.scss`?

But the first question here is: if the code was duplicated in 15.0, what commit did unduplicate it and why?

Reply:
@qsm-odoo i don't know why but this commit unduplicate it.(https://github.com/odoo/odoo/blob/24a4c7f897d4db6bdf0f3cbfc91a3ce7351241a2/addons/mass_mailing/static/src/scss/themes/theme_default.scss).

Reply:
>  I don't know why but this commit unduplicate it.

The first sentence of the commit message seems to explain it? :thinking: See, it was really worth blaming and asking yourself why it was like that before and not anymore :grimacing: 

> The `Styles` option contains styles which are unsupported in
emails. The few that are supported can be replaced by other options.

You [merged that](https://github.com/odoo/odoo/commit/24a4c7f897d4db6bdf0f3cbfc91a3ce7351241a2) apparently @dmo-odoo I guess it's worth reconsidering everything before re-duplicating or sharing :smile:

Reply:
Wow, good catch @qsm-odoo. It's one of the few PRs from designers that ended up reviewed in my team for reasons that aren't always clear to me. Those have often come back to bite us because we had to review them without knowing the full picture etc.

In this particular case, I feel like removing them might have been an oversight. The commit message is saying some features like rounded corners and such are not supported in email clients, but that is not what the scss does. From my understanding it just applies more style to a font-awesome when it has those classes on. I suppose the idea of the commit was to remove this style since the classes needed to enable it would not be available in mass_mailing anymore. However, it seems that they still exist to some extent since those style do activate with these rules.

I feel like at least one of the following options apply:
A. The added styles are not compatible with email clients, in which case it was correct to remove them. We might want to remove them from the templates and thus redesign them a bit.
B. The added styles are compatible with email clients and we still do want them, in which case they should probably be reattached to some other class that is possible to enable in mass_mailing.
C. The added styles are compatible with email clients but we actually don't want them in mass_mailing because they don't really make sense without the special classes that aren't supported in mass_mailing anymore, in which case it is the templates that need to be fixed so that they do not use those classes anymore.

I'd like to summon @Zinston for an opinion here.

Reply:
> In this particular case, I feel like removing them might have been an oversight. The commit message is saying some features like rounded corners and such are not supported in email clients, but that is not what the scss does [...] However, it seems that they still exist to some extent

It's not what the CSS that was removed does indeed: but the commit removes the possibility to apply the classes which are targeted by those removed CSS rules. The fact that they induce rounded corners or shadow is because of bootstrap... So removing those CSS rules seemed legit to me. I feel like we remove the possibility to apply those classes because they don't work in mail clients but kept using them in themes templates (which means those templates do not work in mail clients still)? "However, it seems that they still exist to some extent" -> probably should not if they don't work :grimacing:

> I feel like at least one of the following options apply

I feel that the fact those classes are not possible to apply via options by the user is enough on its own to not use them in mails ever, and thus not have extra CSS targeting them. Code-wise it seems to imply that those classes should be used in templates and can be changed by customers. Combine that with the fact that the base function of those classes (applying shadow, etc) may not work in mail clients: it really does not make sense to use them in templates / in any way.

But I'll let you check and decide from here :+1:

Reply:
I just checked with @Zinston and indeed it's option C. We should remove the uses of those classes in the templates and therefore we don't need to reintroduce the additional style. However, this is stable so we can't easily remove the classes from the templates. What I had in mind is that we could re-introduce the styles in stable but fix the templates on master. Since the PR will be adding them just to remove them afterwards in master, I think duplicating the style would be better than doing a mixin because that way we won't touch any file in website. Does this sound ok for you @qsm-odoo ?

Reply:
Of course :+1:

Reply:
@visp-odoo Sorry for to back and forth on what to do here. We've finally reached a decision:
- For stable versions: duplicate the styles, exactly like you had done the first time.
- In master, do not reintroduce the style but remove the classes from the templates so that the behavior of the template is the same whether website is installed or not.

Since the fix will be completely different in stable than in master, I suggest creating two PRs. We will forward port the stable one up to 16.4 and merge the master one separately. If we don't do that I fear we might forget to make a different fix for master, so I think having two PRs from the start is easier.